### PR TITLE
B 19814 add origin/dest sit auth end date to db

### PR DESCRIPTION
**Pull request by MichaelEngelbertCACI**
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19814)

[integrationPR](https://github.com/transcom/mymove/pull/12593)
## Summary

This Pull request adds the columns origin_sit_auth_end_date and dest_sit_auth_end_date to the mto_shipments table. These dates will be calculated as the day the shipment enters origin/dest SIT + the SIT day allowance (DOFSIT/DDFIST + sit_days_allowance). Storing these values as columns in the database will reduce the need of computing these values on the frontend/backend and instead allow them to be statically held in the db and allow Prime to request an endpoint for the values when needed. 

## Verification Steps for the Author

These are to be checked by the author.

- [v] Tested in the local environment
- [v] Have the Agility acceptance criteria been met for this change?
- [v] Columns are added to the mto_shipments table in the dev_db

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. run git fetch --all in terminal of mymove repo to fetch all updated branches.
2. git checkout integrationTesting
3. git pull
4. git checkout INT-B-19814-add-origin/dest-sit-auth-end-date-to-db to checkout the code from this PR 
5. make db_dev_run
6. make db_dev_fresh db_dev_migrate to run the newly added migration script
7. Once the scripts have ran open up dbeaver and connect to the database and navigate to the mto_shipments table and verify origin_sit_auth_end_date and dest_sit_auth_end_date have been added as columns to the table as a date datatype. 

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Screenshot 2024-05-13 at 11 33 22 AM

### CODE REVIEW

**Comment by deandreJones** [comment]
conflicts

**Review by traskowskycaci — APPROVED** [review]
Some merge conflicts to resolve but otherwise looks good!

**Review by danieljordan-caci — APPROVED** [review]

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Seems to be some funky stuff happening.

Might want to try running this locally and see what happens.

File: migrations/app/migrations_manifest.txt [hunk 1]
```diff
@@ -927,6 +927,7 @@
 20240502175909_add_lookup_valid_columns_to_tac_and_loa_tables.up.sql
 20240502183613_add_support_for_standalone_payment_cap.up.sql
 20240503123556_add_diversion_reason_to_mto_shipments.up.sql
+20240503181242_add_origin_dest_sit_auth_end_date_to_mto_shipments_table
 20240507133524_add_locked_moves_column_to_moves_table.up.sql
```
Comment (cameroncaci) [inline migrations/app/migrations_manifest.txt:931]: Filename is incorrect. It needs the file extension

**Review by cameroncaci — CHANGES_REQUESTED** [review]
Changed from int, hence the breakage

**Review by MichaelEngelbertCACI — COMMENTED** [review]
fixed

**Review by traskowskycaci — CHANGES_REQUESTED** [review]
Seems like some tests are failing. Can you run `make db_dev_fresh` while on this branch and verify that the new columns appear?

**Comment by cameroncaci** [comment]
> fixed

Heads up, it needs the `.up` portion too. Some end in .sql but your file ends in .up.sql

**Comment by danieljordan-caci** [comment]
Looks like all is well. Just renamed the file name in migration manifest. Please re-review gang.

**Review by danieljordan-caci — APPROVED** [review]

**Review by cameroncaci — APPROVED** [review]

**Review by traskowskycaci — APPROVED** [review]

**Review by deandreJones — APPROVED** [review]

**Pull request merged**